Mathlib/Logic/Equiv/Array.lean

```diff
@@ -14,8 +14,20 @@ import Mathlib.Logic.Equiv.List
 namespace Equiv
 
 /-- The natural equivalence between arrays and lists. -/
+@[simps! apply symm_apply]
 def arrayEquivList (α : Type*) : Array α ≃ List α :=
-  ⟨Array.toList, Array.mk, fun _ => rfl, fun _ => rfl⟩
+  ⟨Array.toList, List.toArray, fun _ => rfl, fun _ => rfl⟩
+
+/-- `toVector` and `ofVector` together form an equivalence. -/
+@[simps! apply symm_apply]
+def vectorEquivListVector (α : Type*) (n : ℕ) : Vector α n ≃ List.Vector α n :=
+  ⟨List.Vector.ofVector, List.Vector.toVector, fun _ => rfl, fun _ => rfl⟩
+
+theorem toList_vectorEquivListVector_apply {α : Type*} {n : ℕ} (v : Vector α n) :
+  (vectorEquivListVector α n v).toList = arrayEquivList α v.toArray := rfl
+
+theorem toArray_vectorEquivListVector_symm_apply {α : Type*} {n : ℕ} (v : List.Vector α n) :
+  ((vectorEquivListVector α n).symm v).toArray = (arrayEquivList α).symm v.toList := rfl
```

Comment:
I don't think you need these, since neither side is simp-normal anyway.

Reply:
OK.